### ISSUE TITLE
[Bug] Resx files don't work with Android App Bundle

### ISSUE THREAD

**Issue by DiamondKesha** [xamarin/xamarin-android#3596]
### Description
If use Android App Bundle, then AppResource.de.resx and other don't work, but AppResource.resx work.
I use - [this example](https://docs.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/usingresxlocalization/)
### Steps to Reproduce
1. Add <**AndroidPackageFormat>aab</AndroidPackageFormat**> in project.csproj

### Expected Behavior
Changing the language in the program

### Actual Behavior
The language in the program should change, but it does not change.

### Basic Information
- Platform Target Frameworks: 5.0
- Affected Devices: Samsung

**Comment by losmacko** [comment]
I have the same issue. Any workaround?

**Comment by jonathanpeppers** [comment]
I did not test this, but a theoretical workaround, would be to set `.dll` for the `AndroidStoreUncompressedFileExtensions` setting:

![image](https://user-images.githubusercontent.com/840039/64556779-bd8de100-d305-11e9-9e0a-45c8bfe5b465.png)

**Comment by EmilAlipiev** [comment]
Does this solution have any side effects? is it not increasing bundle size? if i am not mistaken dlls were compressed into apk but now not in aab. is it not a step back?

**Comment by jonathanpeppers** [comment]
DLLs are not compressed, because [mmap](https://en.wikipedia.org/wiki/Mmap) is used to load them into memory directly from the APK.

I think we looked at allowing them .NET assemblies to be compressed at one point, but it hurt startup performance.

**Comment by brendanzagaeski** [comment]
*Release status update*

A new Preview version has now been published that includes the fix for this item. The fix is not yet included in a Release version. I will update this item again when a Release version is available that includes the fix.

The fix is not yet available on macOS. I will update this item again when a Preview version with the fix is available on macOS.

Fix included in Xamarin.Android 10.0.99.100.

Fix included on Windows in Visual Studio 2019 version 16.4 Preview 1. To try the Preview version that includes the fix, [check for the latest updates](https://docs.microsoft.com/visualstudio/install/update-visual-studio?view=vs-2019) in [Visual Studio Preview](https://visualstudio.microsoft.com/vs/preview/).

Fix not yet available on macOS.

**Comment by thisisthekap** [comment]
@brendanzagaeski As we are quite desperately looking forward to AAB publishing (mainly because of download sizes), we would appreciate it to see this issue in a hotfix Version of Xamarin.Android,  as this issue is also listed in the "Known Issues" section of Xamarin.Android 10.0.0.43.

**Comment by jonathanpeppers** [comment]
@thisisthekap did the workaround work? https://github.com/xamarin/xamarin-android/issues/3596#issuecomment-529607832

**Comment by lahna200** [comment]
I Confirm, the workaround works, mp3 files must also be included to work properly.

**Comment by jonathanpeppers** [comment]
@lahna200 there are 2 fixes for app bundles & compression in the next release (Xamarin.Android 10.1), but I bet this one will fix the mp3 issue: https://github.com/xamarin/xamarin-android/pull/3604

**Comment by brendanzagaeski** [comment]
*Release status update*

A new Preview version has now been published for macOS that includes the fix for this item.

Fix included in Xamarin.Android 10.1.0.1.

Fix included on macOS in Visual Studio 2019 for Mac version 8.4 Preview 1. To try the Preview version that includes the fix, check for the latest updates on the **Preview** [updater channel](https://docs.microsoft.com/visualstudio/mac/update).

**Comment by EmilAlipiev** [comment]
Is there already a documentation or article about how to achieve app bundles step by step?

**Comment by andrewleader** [comment]
I'm also blocked on this, eagerly waiting for the stable channel release fix to hit Azure DevOps so I can build in the cloud again! Building locally using the preview in the meantime.

**Comment by jonathanpeppers** [comment]
@andrewleader there is a workaround for this one: https://github.com/xamarin/xamarin-android/issues/3596#issuecomment-529607832

**Comment by brendanzagaeski** [comment]
*Release status update*

A new Release version has now been published on Windows that includes the fix for this item. The fix is not yet published in a Release version on macOS. I will update this item again when a Release version is available on macOS that includes the fix.

Fix included in Xamarin.Android 10.1.0.30.

Fix included on Windows in Visual Studio 2019 version 16.4. To get the new version that includes the fix, [check for the latest updates](https://docs.microsoft.com/visualstudio/install/update-visual-studio?view=vs-2019) or install the latest version from <https://visualstudio.microsoft.com/downloads/>.

(Fix also included on macOS in Visual Studio 2019 for Mac version 8.4 Preview 2.1 and higher. To try the Preview version that includes the fix, check for the latest updates on the **Preview** [updater channel](https://docs.microsoft.com/visualstudio/mac/update).)

**Comment by brendanzagaeski** [comment]
*Release status update*

A new Release version has now been published on macOS that includes the fix for this item.

Fix included in Xamarin.Android 10.1.1.0.

Fix included on macOS in Visual Studio 2019 for Mac version 8.4. To get the new version that includes the fix, check for the latest updates on the **Stable** [updater channel](https://docs.microsoft.com/visualstudio/mac/update).

(Fix also included on Windows in Visual Studio 2019 version 16.4 and higher. To get the new version that includes the fix, [check for the latest updates](https://docs.microsoft.com/visualstudio/install/update-visual-studio?view=vs-2019) or install the latest version from <https://visualstudio.microsoft.com/downloads/>.)

**Comment by acuntex** [comment]
@brendanzagaeski Does either not work or is a regressed bug.

When starting the app in a German Simulator all texts show correctly in German.
When bundling as aab and uploading to the store all texts are again English on a German device.

Anyone else experiencing this?

**Comment by brendanzagaeski** [comment]
@acuntex, thanks for the heads-up! If are using Xamarin.Android 10.3 in Visual Studio 2019 version 16.6 or Visual Studio 2019 for Mac version 8.6, one likely possible cause is  <https://github.com/xamarin/xamarin-android/issues/4664>. If the workaround mentioned in that item does not resolve the issue for your project, then if you might be able to submit a new issue that has the [diagnostic MSBuild output](https://docs.microsoft.com/xamarin/android/troubleshooting/troubleshooting#diagnostic-msbuild-output) from your project zipped and attached so the team can take a look, that would be perfect. Thanks in advance!

**Comment by acuntex** [comment]
I can confirm that the workaround from #4664 worked.
Thanks!
